wallet/wallet.go

```diff
@@ -152,6 +172,30 @@ func (w *Wallet) NewKeyInfo() (*types.KeyInfo, error) {
 	return w.keyInfoForAddr(newAddr)
 }
 
+// CreateTicket computes a valid ticket.
+// 	params:  proof  []byte, minerAddress address.Address.
+//  returns:  types.Signature ( []byte ), error
+func (w *Wallet) CreateTicket(proof proofs.PoStProof, signerPubKey []byte) (types.Signature, error) {
```

Comment:
It seems like this function belongs on an abstraction on top of wallet.  All of its operations depend on public methods of the wallet, and creating a ticket seems far enough removed from a wallet that another abstraction is sensible.

For the record I have a similar opinion about the `SignBytes` method which is well established so I might just be wrong here.

Reply:
@ZenGround0 check out the last commit.  A bunch of things were cleaned up so this should look better to you now.